src/amp_local_goal/amp_local_goal/follow_waypoint.py

```diff
@@ -0,0 +1,196 @@
+import rclpy
+import numpy as np
+from rclpy.node import Node
+from nav2_msgs.msg import Costmap
+from nav_msgs.msg import Odometry
+import cv2
+from geometry_msgs.msg import Quaternion, PoseStamped, PoseWithCovarianceStamped
+import scipy.ndimage
+from transforms3d.euler import quat2euler
+
+# Costmap values (hardcoded in ROS)
+LETHAL_OBSTACLE = 254
+NO_INFO = 255
+INFLATION_BUFFER = 253
+MEDIUM_COST = 128
+FREE_SPACE = 0
+
+# Remapping costmap values to custom brightness values in costmap image
+COST_IMAGE_MAPPINGS = {
+    LETHAL_OBSTACLE: 255,
+    NO_INFO: 0,
+    INFLATION_BUFFER: 128,
+    MEDIUM_COST: 192,
+    FREE_SPACE: 0
+}
+
+
+class CostMapSubscriber(Node):
+
+    initialKartYaw = 0
+    currentKartYaw = 0
+    currentKartQuat = Quaternion(x=0.0, y=0.0, z=0.0, w=1.0)
+    parabola_numer = 0
+    parabola_denom = 0
+    show_costmaps = False
+
+    goal_x = 0
+    goal_y = 0
+    goal_z = 0
+
+    def __init__(self):
+        super().__init__('CostMap_Subscriber')
```

Comment:
```suggestion
        super().__init__('costmap_subscriber')
```
Bit nitpicky, but ROS convention is to keep node names in snake case. Might also consider renaming it to something more descriptive like `find_local_goal`